stdlib/out_channel.ml

```diff
@@ -0,0 +1,62 @@
+(**************************************************************************)
+(*                                                                        *)
+(*                                 OCaml                                  *)
+(*                                                                        *)
+(*             Xavier Leroy, projet Cristal, INRIA Rocquencourt           *)
+(*                                                                        *)
+(*   Copyright 2019 Institut National de Recherche en Informatique et     *)
+(*     en Automatique.                                                    *)
+(*                                                                        *)
+(*   All rights reserved.  This file is distributed under the terms of    *)
+(*   the GNU Lesser General Public License version 2.1, with the          *)
+(*   special exception on linking described in the file LICENSE.          *)
+(*                                                                        *)
+(**************************************************************************)
+
+
+type t = out_channel
+
+let stdout = stdout
+let stderr = stderr
+
+type open_flag =
+  | Append
+  | Create of int
+  | Exclusive
+  | Truncate
+  | Binary
+  | Non_blocking
+
+let open_ ?flags:(flags0 = []) path =
+  let aux (flags, perm) = function
+    | Append -> Open_append :: flags, perm
+    | Create perm -> Open_creat :: flags, perm
+    | Exclusive -> Open_excl :: flags, perm
+    | Truncate -> Open_trunc :: flags, perm
+    | Binary -> Open_binary :: flags, perm
+    | Non_blocking -> Open_nonblock :: flags, perm
+  in
+  let flags, perm = List.fold_left aux ([Open_wronly], 0o666) flags0 in
```

Comment:
Is `0o666` really the right default ? I would rather go for `0o644`.

Reply:
OK, fixed.

Reply:
On the other hand `0o666` is the one used by `Stdlib.open_out`, so there is a slight potential for confusion here.

Reply:
The permissions given to `open` are modified by the user's umask.  `0o666` is the traditional "friendly" set of permissions.  Most users have a `0o022` umask that will turn write permissions off for group and for others, resulting in the expected `0o644` permissions.  But some users want their files to be group-writable by default, and have `0o002` umask, and there is no reason to annoy them by forcing the permissions to `0o644`.

Reply:
Makes sense, thanks for the explanation. Will set the default back to `0o666`.